etl/types/types.go

```diff
@@ -29,13 +29,14 @@ type Section interface {
 func (m MainSection) IsSection() bool { return true }
 
 type MainSection struct {
+	Section
```

Comment:
"recursive" Embeding.
You embeding a reference to "Section typed struct" to a struct.
Please rethink.